cli/flox/src/commands/containerize/mod.rs

```diff
@@ -64,17 +64,17 @@ impl Containerize {
 
         let built_environment = env.build(&flox)?;
         let env_name = env.name();
+        let manifest = env.lockfile(&flox)?.manifest;
+        let mode = manifest.activate.mode.unwrap_or_default();
 
         let source = if std::env::consts::OS == "linux" {
-            let container_config = env
-                .lockfile(&flox)?
-                .manifest
+            let container_config = manifest
                 .containerize
                 .and_then(|c| c.config)
                 .map(|c| c.into());
             // this method is only executed on linux
             #[cfg_attr(not(target_os = "linux"), allow(deprecated))]
-            let builder = MkContainerNix::new(built_environment.develop, container_config);
+            let builder = MkContainerNix::new(built_environment.for_mode(&mode), container_config);
```

Comment:
blocking: I think we need to pass `--mode {mode}` in `mkContainer.nix`

Reply:
Thanks! Added in a fixup commit. Worth noting that CI won't exercise it for macOS until after https://github.com/flox/flox/pull/2769 but again it shouldn't matter because we're not passing the arguments over the proxy container boundary.